tests/integration/bundle/multi_cloud.py

```diff
@@ -66,26 +69,25 @@ def juju(self, *args, json_output=True) -> dict:
 
     def get_controller_name(self, cloud: str) -> Optional[str]:
         """Gets controller name for specified cloud, e.g. localhost, microk8s, lxd, etc."""
-        res = self.juju("controllers")
+        res = self.juju_cli("controllers")
         for controller in res.get("controllers", {}):
             if res["controllers"][controller].get("cloud") == cloud:
                 return controller
 
         return None
 
-    async def get_model(self, controller_name: str, model_name: str) -> Model:
-        """Gets a juju model on specified controller, raises JujuConnectionError if not found."""
-        state = await OpsTest._connect_to_model(controller_name, model_name)
-        return state.model
+    def get_model(self, controller_name: str, model_name: str) -> str | None:
+        """Gets a juju model on specified controller, and None if not found."""
+        try:
+            self.juju.add_model(model=model_name, controller=controller_name)
+            return None
+        except jubilant.CLIError:
+            # model exists!
+            return model_name
```

Comment:
I think the names of these two methods (`get_model`, `get_or_create_model`) should be changed to better reflect what they are doing. This one is creating a model, not connecting like before. I think it also makes the next method redundant. So maybe just have one helper?